src/app/pages/wallets/management/pages-wallets-management.component.ts

```diff
@@ -55,59 +48,34 @@ export class PagesWalletsManagementComponent implements OnInit {
     });
   }
 
-  public createWallet({ name }: IWalletModalData): void {
-    this.myWalletsService.createWallet(WalletsManagementItem.create({ name })).subscribe({
-        next: (wallet: WalletsManagementItem) => {
-          this.myWalletsData.data = [wallet, ...this.myWalletsData.data!];
-          this.systemNotificationsService.showNotification({ message: 'Congratulations! Your wallet was created successfully.' });
-        },
-        error: () => {
-          this.systemNotificationsService.showNotification({ message: 'Sorry. Something went wrong and your wallet was not saved. Contact administrator.' });
-        },
-    });
-  }
-
-  public updateWallet({ id }: WalletsManagementItem, { name }: IWalletModalData): void {
-    this.myWalletsService.updateWallet(WalletsManagementItem.create({ id: id!, name })).subscribe({
-        next: (updatedWallet: WalletsManagementItem) => {
-            this.myWalletsData.data = this.myWalletsData.data!.map(walletItem => {
-                if (walletItem.id === id) {
-                    return updatedWallet;
-                }
-                return walletItem;
-            });
-        },
-        error: () => {
-            this.systemNotificationsService.showNotification({ message: 'Some server error during updating' });
-        },
-    });
+  public handleWalletCreate(): void {
+    this.pagesWalletsManagementEditorService.openWalletEditor().subscribe({
+      next: (createdName: WalletsManagementItem | null) => {
+        createdName && this.createWallet(createdName);
+      },
+    })
   }
 
-  public handleWalletCreate(): void {
-    this.openWalletModal().subscribe( (walletModalData?: IWalletModalData) => {
-            if(walletModalData) {
-                this.createWallet(walletModalData);
-            }
-        },
-    );
+  private createWallet(wallet: WalletsManagementItem): void {
+    this.myWalletsData.data = [wallet, ...this.myWalletsData.data!];
   }
 
   public handleWalletEdit(wallet: WalletsManagementItem): void {
-    this.openWalletModal(wallet).subscribe( (walletModalData?: IWalletModalData) => {
-            if (walletModalData === undefined) {
-              return;
-            }
-            this.updateWallet(wallet, walletModalData);
-        },
-    );
+    this.pagesWalletsManagementEditorService.openWalletEditor(wallet).subscribe({
+      next: (createdName: WalletsManagementItem | null) => {
```

Comment:
This is not created name but wallet item. Also please use `if` instead magic below